internal/queryexecutor/executor.go

```diff
@@ -15,26 +16,90 @@ import (
 // WorldStateQueryExecutor executes a given set of query criterias on the states stored in
 // the world state database and returns a set of keys whose values are matching the given
 // criterias
-type WorldStateQueryExecutor struct {
+type WorldStateJSONQueryExecutor struct {
 	db     worldstate.DB
 	logger *logger.SugarLogger
 }
 
-func NewWorldStateQueryExecutor(db worldstate.DB, l *logger.SugarLogger) *WorldStateQueryExecutor {
-	return &WorldStateQueryExecutor{
+func NewWorldStateJSONQueryExecutor(db worldstate.DB, l *logger.SugarLogger) *WorldStateJSONQueryExecutor {
+	return &WorldStateJSONQueryExecutor{
 		db:     db,
 		logger: l,
 	}
 }
 
+func (e *WorldStateJSONQueryExecutor) ExecuteQuery(dbName string, selector []byte) (map[string]bool, error) {
```

Comment:
I would be helpful to document here the structure of the json object expected to be in `selector`. That is, unless you plan to document it in a later PR on a higher layer. It is a map, insid a map, inside a map, not a trivial structure.... it is ok to do this in another PR.